src/sparseml/modifiers/obcq/utils/helpers.py

```diff
@@ -22,22 +23,33 @@
 
 
 class Catcher(torch.nn.Module):
-    def __init__(self, module, target_keys):
+    def __init__(self, module):
         super().__init__()
         self.module = module
-        self.cache = {key: [] for key in target_keys}
-        self.target_keys = target_keys
+        self.target_keys = None
+        self.cache = defaultdict(list)
         self.cache["inputs"] = []
 
     def forward(self, *args, **kwargs):
         self.cache["inputs"].append(args)
+        if self.target_keys is None:
+            self.target_keys = self._get_target_keys(kwargs.keys())
+
         for key in self.target_keys:
             self.cache[key].append(kwargs[key])
         raise ValueError
 
     def get_cache(self):
         return self.cache
 
+    def _get_target_keys(self, input_keys):
```

Comment:
I suggest creating a list of _DEFAULT_TARGET_KEYS = ["attention_mask", "position_ids"] at the top of helpers.py to make this more explicit in case this needs to change in the future.